// webgl-test-shared/lib/biomes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Tile_1 = require("./Tile");
var BIOMES = [
    {
        name: "magmaFields",
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.lava,
                    isWall: false
                },
                minWeight: 0.2,
                minDist: 3
            },
            {
                info: {
                    type: Tile_1.TileType.magma,
                    isWall: false
                }
            }
        ]
    },
    {
        name: "mountains",
        generationInfo: {
            minHeight: 0.8
        },
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.rock,
                    isWall: true
                },
                minWeight: 0.8,
                minDist: 2
            },
            {
                info: {
                    type: Tile_1.TileType.rock,
                    isWall: false
                }
            }
        ]
    },
    {
        name: "tundra",
        generationInfo: {
            maxTemperature: 0.3,
            maxHumidity: 0.5
        },
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.rock,
                    isWall: true
                },
                minWeight: 0.8,
                minDist: 4
            },
            {
                info: {
                    type: Tile_1.TileType.ice,
                    isWall: false
                },
                minWeight: 0.6,
                minDist: 3
            },
            {
                info: {
                    type: Tile_1.TileType.snow,
                    isWall: false
                }
            }
        ]
    },
    {
        name: "desert",
        generationInfo: {
            minTemperature: 0.7,
            maxHumidity: 0.5
        },
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.sandstone,
                    isWall: true
                },
                minWeight: 0.6,
                minDist: 2
            },
            {
                info: {
                    type: Tile_1.TileType.sandstone,
                    isWall: false
                },
                minWeight: 0.5,
                minDist: 1
            },
            {
                info: {
                    type: Tile_1.TileType.sand,
                    isWall: false
                }
            }
        ]
    },
    {
        name: "swamp",
        generationInfo: {
            minTemperature: 0.7,
            minHumidity: 0.7
        },
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.dirt,
                    isWall: false
                },
                minWeight: 0.5,
                minDist: 2
            },
            {
                info: {
                    type: Tile_1.TileType.sludge,
                    isWall: false
                }
            }
        ]
    },
    {
        name: "grasslands",
        generationInfo: {},
        tiles: [
            {
                info: {
                    type: Tile_1.TileType.grass,
                    isWall: false
                }
            }
        ]
    }
];
exports.default = BIOMES;
